pkg/agent/core/state.go

```diff
@@ -686,6 +693,44 @@ func (s *state) desiredResourcesFromMetricsOrRequestedUpscaling(now time.Time) (
 		memGoalCU := uint32(memGoalBytes / s.Config.ComputeUnit.Mem)
 
 		goalCU = util.Max(cpuGoalCU, memGoalCU)
+
+		// For LFC metrics, if enabled:
+		if *s.scalingConfig().EnableLFCMetrics {
+			cfg := s.scalingConfig()
+			wssValues := s.LFCMetrics.ApproximateworkingSetSizeBuckets
+			// At this point, we can assume that the values are equally spaced at 1 minute apart,
+			// starting at 1 minute.
+			offsetIndex := *cfg.LFCMinWaitBeforeDownscaleMinutes - 1 // -1 because values start at 1m
+			windowSize := *cfg.LFCWindowSizeMinutes
+			// Handle invalid metrics:
+			if len(wssValues) < offsetIndex+windowSize {
+				s.warn("not enough working set size values to make scaling determination")
+			} else {
+				estimateWss := EstimateTrueWorkingSetSize(wssValues, WssEstimatorConfig{
+					MaxAllowedIncreaseFactor: 3.0, // hard-code this for now.
+					InitialOffset:            offsetIndex,
+					WindowSize:               windowSize,
+				})
+				projectLen := 0.5 // hard-code this for now.
+				predictedHighestNextMinute := ProjectNextHighest(wssValues, projectLen, estimateWss)
```

Comment:
WDYT about filtering `wssValue` here for `< estimateWss`, so that you don't need to pass estimateWss, simplifying the semantics of ProjectNextHighest?

Reply:
Fair enough, yeah - implemented in a96af7b0c8985bf8381becb2c398d90fe9f77507